crates/turbopack-ecmascript/src/analyzer/top_level_await.rs

```diff
@@ -1,19 +1,24 @@
-use swc_core::ecma::{
-    ast::*,
-    visit::{noop_visit_type, Visit, VisitWith},
+use swc_core::{
+    common::Span,
+    ecma::{
+        ast::*,
+        visit::{noop_visit_type, Visit, VisitWith},
+    },
 };
 
-pub(crate) fn has_top_level_await(m: &Program) -> bool {
+/// Checks if the program contains a top level await, if it does it will returns
+/// the span of the first await.
+pub(crate) fn has_top_level_await(m: &Program) -> Option<Span> {
     let mut visitor = TopLevelAwaitVisitor::default();
 
     m.visit_with(&mut visitor);
 
-    visitor.has_top_level_await
+    visitor.top_level_await_span
 }
 
 #[derive(Default)]
 struct TopLevelAwaitVisitor {
```

Comment:
It might be possible to merge this visitor into `detect_dynamic_export` to avoid walking the ast multiple times